### PR TITLE
Making tests work

### PR DESCRIPTION
**Pull request by lucasjose501**
This PR prepares the server code for testing and future development. Also fixing current tests.

You can test the application using the console command 'php artisan test'. Current code coverage is 53% and I'll be working on this soon.
Some tests are skipped because they are not implemented yet.

### CODE REVIEW

**Comment by lucasjose501** [comment]
Now that I saw #99 I realized that the forgot-password tests can not be implemented, and I'll remove the file completely because it's not necessary.

**Comment by bimbashrestha** [comment]
Hey @lucasjose501 at a quick glance, the changes look good. Thanks for making the PR!

I'm realizing I need to sit down and think a bit more carefully about how to test Blobbackup. I'll come back to this once I've done that. Sorry for the delay btw.

**Comment by lucasjose501** [comment]
Here an example: https://gist.github.com/lucasjose501/e1a561f5b4ce4e66bf2ac65a896ee824

File: server/database/factories/UserFactory.php [hunk 1]
```diff
@@ -15,10 +15,9 @@ class UserFactory extends Factory
     public function definition()
     {
         return [
-            'name' => $this->faker->name(),
             'email' => $this->faker->unique()->safeEmail(),
             'email_verified_at' => now(),
-            'password' => '$2y$10$92IXUNpkjO0rOQ5byMi.Ye4oKoEa3Ro9llC/.og/at2.uheWG/igi', // password
+            'password' => bcrypt('password'),
```
Comment (bimbashrestha) [inline server/database/factories/UserFactory.php:20]: Nice!

**Review by bimbashrestha — APPROVED** [review]
Hey @lucasjose501 sorry for the delay in reviewing this! Was a little caught up with some other tasks.

The changes look great! I tested your PR and I see that `./vendor/bin/phpunit` passes now:)

>Here an example: https://gist.github.com/lucasjose501/e1a561f5b4ce4e66bf2ac65a896ee824

I like this workflow. Do you mind including your yml file inside the `.github/workflows` folder in this PR? I'd suggest having tests run on `push` and `pull_request` to the `dev` branch for the time being.

Everything else looks good to me. I'll merge after we add a `yml` file.

**Comment by lucasjose501** [comment]
Not sure if this will work now because of the folder structure in the repo

**Comment by bimbashrestha** [comment]
>Not sure if this will work now because of the folder structure in the repo.

Looks like it worked: https://github.com/Blobbackup/Blobbackup/runs/5797062568?check_suite_focus=true. Nicely done! 

>Edit: Also, I'm not expert in PRs on github but I'll make future PRs in separated branches instead of dev, sorry for that.

I think it's perfectly fine to make PRs directly to `dev` for changes that are unlikely to affect the current production service (testing qualifies!).

Thanks @lucasjose501!

**Comment by lucasjose501** [comment]
That was smooth haha, I hope I got it right and will trigger only for changes in the 'server' directory.

**Pull request merged**